File: frontend/src/types/product-improvement.ts
export interface ImprovementSuggestion {
  id: string;
  suggestion_key: string;
  category: string;
  title: string;
  description: string;
  priority: number;
  severity: string;
  confidence: string;
  evidence_count: number;
  frequency_rate: string;
  evidence_review_ids: { items?: string[] };
  expected_impact: Record<string, unknown> | null;
  status: "PROPOSED" | "ACCEPTED" | "IGNORED";
}

export interface ImprovementReport {
  id: string;
  review_analysis_result_id: string;
  source_product_id: string;
  version: number;
  algorithm_version: string;
  status: string;
  source_type: string;
  is_mock_data: boolean;
  data_sources: Record<string, unknown>;
  input_conditions: Record<string, unknown>;
  summary: {
    sample_size?: number;
    suggestion_count?: number;
    limitations?: string[];
  };
  created_at: string;
  suggestions: ImprovementSuggestion[];
}

export interface ConfirmationResult {
  id: string;
  operation_type?: string;
  status: string;
  risk_warning: string | null;
  execution_result: Record<string, unknown> | null;
}

export interface ImprovementDraftItem {
  title: string;
  description: string;
}

export interface ImprovementDraftVersion {
  id: string;
  content_id: string;
  source_product_id: string;
  site: string;
  version: number;
  sequence: number;
  status: string;
  report_id: string | null;
  items: ImprovementDraftItem[];
  change_summary: string;
  created_at: string;
}

export interface ImprovementDraftList {
  items: ImprovementDraftVersion[];
  total: number;
}

File: backend/src/sellpilot/schemas/product_improvement.py
from datetime import datetime
from decimal import Decimal
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from sellpilot.schemas.confirmation import ConfirmationTaskResponse


class ImprovementGenerateRequest(BaseModel):
    analysis_id: UUID
    force_regenerate: bool = False


class ImprovementSuggestionResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    suggestion_key: str
    category: str
    title: str
    description: str
    priority: int
    severity: Decimal
    confidence: Decimal
    evidence_count: int
    frequency_rate: Decimal
    evidence_review_ids: dict
    expected_impact: dict | None
    status: str


class ImprovementReportResponse(BaseModel):
    id: UUID
    review_analysis_result_id: UUID
    source_product_id: str
    version: int
    algorithm_version: str
    status: str
    source_type: str
    is_mock_data: bool
    data_sources: dict
    input_conditions: dict
    summary: dict
    created_at: datetime
    suggestions: list[ImprovementSuggestionResponse]


class SuggestionUpdateRequest(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=255)
    description: str | None = Field(default=None, min_length=1, max_length=4000)
    status: str | None = Field(default=None, pattern="^(PROPOSED|ACCEPTED|IGNORED)$")


class ImprovementDraftRequest(BaseModel):
    idempotency_key: str = Field(min_length=8, max_length=128)
    site: str = Field(default="sg", min_length=2, max_length=16)
    target_language: Literal["und"] = "und"
    suggestion_ids: list[UUID] = Field(min_length=1, max_length=50)


class ImprovementDraftResponse(ConfirmationTaskResponse):
    pass


class ImprovementDraftItem(BaseModel):
    title: str = Field(min_length=1, max_length=255)
    description: str = Field(min_length=1, max_length=4000)


class ImprovementDraftVersionResponse(BaseModel):
    id: UUID
    content_id: UUID
    source_product_id: str
    site: str
    version: int
    sequence: int
    status: str
    report_id: UUID | None
    items: list[ImprovementDraftItem]
    change_summary: str
    created_at: datetime


class ImprovementDraftListResponse(BaseModel):
    items: list[ImprovementDraftVersionResponse]
    total: int


class ImprovementDraftRevisionRequest(BaseModel):
    idempotency_key: str = Field(min_length=8, max_length=128)
    expected_version: int = Field(ge=1)
    items: list[ImprovementDraftItem] = Field(min_length=1, max_length=50)


class ImprovementDraftHistoryClearRequest(BaseModel):
    idempotency_key: str = Field(min_length=8, max_length=128)
    source_product_id: str = Field(min_length=1, max_length=100)
